README.md

```diff
@@ -549,6 +550,61 @@ Medium.
 - [Idea discussion](https://rust-lang.zulipchat.com/#narrow/channel/421156-gsoc/topic/Idea.3A.20Port.20.60std.3A.3Aarch.60.20test.20suite.20to.20.60rust-lang.2Frust.60)
 - [t-libs/stdarch](https://rust-lang.zulipchat.com/#narrow/channel/208962-t-libs.2Fstdarch)
 
+### Distributed and focused verification
+
+**Description**
+
+We have taken a goal to
+[instrument the Rust standard library with safety contracts](https://rust-lang.github.io/rust-project-goals/2025h1/std-contracts.html).
+With this approach we are moving from informal comments specifying safety
+requirements on `unsafe` functions to executable Rust code.
+To prove that these safety requirements hold in all possible executions we need
+to run static analysis tools.
+We would further like to prove that these safety requirements continue to hold
+despite code changes.
+Therefore, we will want to run static analysis tools that prove these safety
+requirements in continuous integration.
+We have started such an effort in our
+[verify-rust-std](https://github.com/model-checking/verify-rust-std) fork.
+
+Static analysis tools will typically require compute effort that may go up to
+minutes per function that needs to be proved.
+With a growing number of contracts we will, therefore, need to distribute proof
+effort across multiple nodes.
+We may, however, also _reduce_ the proof effort when we can identify that any
+given proof necessarily continues to hold as the transitive closure of functions
+goverened by the contract has not been modified since the proof last completed.
+
+We are looking for contributions in the following areas:
+1. A reachability- or impact analysis that, given a code change, determines
```

Comment:
That would be awesome. The one thing to keep in mind is that the optimal solution may vary depending on the verification strategy.

Reply:
Added more notes to this effect.